templates/cluster-template-aks.yaml

```diff
@@ -65,6 +65,7 @@ metadata:
   name: agentpool0
   namespace: default
 spec:
+  maxPods: 24
```

Comment:
is this value for testing? should we leave it empty if it's not required and let the default be whatever AKS defaults to?

Reply:
It's for testing, yes

Reply:
can we remove it now that the tests have run? or do we want to add them to the test template specifically but not the reference template maybe?

Reply:
I have a slight preference for keeping them in to get regular E2E validation on this configuration input. If it breaks existing E2E patterns, happy to remove.

Reply:
if the goal is to represent a best practice value to users in our published release flavor, it should be in here. If the goal is just to test some arbitrary value regularly in e2e so the functionality doesn't regress, it should be a patch in https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/templates/test/ci/prow-aks-multi-tenancy so that it only gets added to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/test/ci/cluster-template-prow-aks-multi-tenancy.yaml

Reply:
Done in latest change.